operator/src/main/java/oracle/kubernetes/operator/helpers/HttpClientPool.java

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) 2017, 2020, Oracle Corporation and/or its affiliates.
+// Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+
+package oracle.kubernetes.operator.helpers;
+
+import java.net.http.HttpClient;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.SynchronousQueue;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.ThreadPoolExecutor;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.concurrent.atomic.AtomicReference;
+
+import oracle.kubernetes.operator.logging.LoggingFacade;
+import oracle.kubernetes.operator.logging.LoggingFactory;
+import oracle.kubernetes.operator.logging.MessageKeys;
+import oracle.kubernetes.operator.work.Container;
+import oracle.kubernetes.operator.work.ContainerResolver;
+
+public class HttpClientPool extends Pool<HttpClient> {
+  private static final LoggingFacade LOGGER = LoggingFactory.getLogger("Operator", "Operator");
+  private static final HttpClientFactory FACTORY = new DefaultClientFactory();
+  private static HttpClientPool SINGLETON = new HttpClientPool();
+  private static ThreadFactory threadFactory;
+  private final AtomicBoolean isFirst = new AtomicBoolean(true);
+
+  // HttpClient instance that will be shared
+  private final AtomicReference<HttpClient> instance = new AtomicReference<>();
+
+  public static void initialize(ThreadFactory threadFactory) {
+    HttpClientPool.threadFactory = threadFactory;
+  }
+
+  private static Runnable wrapRunnable(Runnable r) {
+    return new Runnable() {
+      @Override
+      public void run() {
+        try {
+          r.run();
+        } catch (Throwable t) {
+          // These will almost always be spurious exceptions
+          LOGGER.finer(MessageKeys.EXCEPTION, t);
+        }
+      }
+    };
+  }
+
+  public static HttpClientPool getInstance() {
+    return SINGLETON;
+  }
+
+  @Override
+  protected HttpClient create() {
+    return instance.updateAndGet(
```

Comment:
Trying to figure out what you're doing here.  I thought you wanted to pool HttpClient instances, but you are creating a single instance and returning it every time.

Reply:
I was modeling this based on the existing ClientPool class implementation in helper package.  JDK11 http client and request objects are thread-safe and can be safely shared. So I thought creating a single instance and sharing it every time should be fine. In my testing, it seem to work fine. Please let me know if you see any issues with this or if any changes are needed. Thanks.

Reply:
As above, let's rid of this class.

Reply:
Removed in commit ba303f0. Thanks.